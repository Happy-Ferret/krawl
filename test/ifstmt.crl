import "stdio.h";

func main(argc int, argv **byte) int
{
	big, small := 100, 1;

	if big > small {
		stdio.printf("big > small\n");
	}

	if small > big {} else {
		stdio.printf("!(small > big)\n");
	}

	if big == small {
		stdio.printf("true\n");
	} else {
		stdio.printf("false\n");
	}

	if big != small {
		stdio.printf("true\n");
	} else {
		stdio.printf("false\n");
	}

	if big != 0 {
		if small == 1 {
			stdio.printf("all ok\n");
		}
	} else {
	}

	if big == -1 { stdio.printf("omg, no!\n"); } else {
		if small != 15 {
			stdio.printf("interesting\n");
		}

		if small != 1 {} else {
			stdio.printf("very interesting\n");
		}
	}

	a, b, c := 10, 20, 30;
	if big > small {
		a, b, (c) := 1, 2, 3;
		stdio.printf("%d %d %d\n", a, b, c);
	}
	stdio.printf("%d %d %d\n", a, b, c);

	return 0;
}
