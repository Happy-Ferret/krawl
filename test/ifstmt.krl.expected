big > small
!(small > big)
false
true
all ok
interesting
very interesting
1 2 3
10 20 3
